### PR TITLE
8261 - Avoid including Cesium3DTileSet in Picking.js

### LINKED ISSUE #8261: Avoid importing Cesium3DTileset in Picking.js
Since updating to the latest version of Cesium, I noticed my project's bundle size had grown 85% by over 850kb minified.

I tracked down a large chunk of it to a single `instanceof Cesium3DTileset` use here: https://github.com/AnalyticalGraphicsInc/cesium/blob/d26ccfea38dcf4623cf5da3faca141913c64cfe1/Source/Scene/Picking.js#L549 causing the bundle size to grow by 549 kB, while I do not use `Cesium3DTileset` in my project.

Perhaps instead of using `instanceof` we could have a static `type` enum property to check against? For example `if (primitive.type === Type.Cesium3DTileset)`

### PR DESCRIPTION
**Pull request by Samulus**
Fixes #8261

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thank you so much for the pull request @Samulus! I noticed this is your first pull request and I wanted to say welcome to the Cesium community!

The [Pull Request Guidelines](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines) is a handy reference for making sure your PR gets accepted quickly, so make sure to skim that.

* :x: Missing `CONTRIBUTORS.md` entry.
    * Please add yourself to the [contributors](https://github.com/AnalyticalGraphicsInc/cesium/blob/8261/instance_of_picking/CONTRIBUTORS.md) file!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/8261/instance_of_picking/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

**Comment by hpinkos** [comment]
Can we do something similar to avoid sucking in `Cesium3DTileFeature` too?

**Comment by lilleyse** [comment]
@hpinkos `Cesium3DTileFeature` doesn't have too many imports and will unlikely have more. It might be ok to skip it.

```js
import Color from '../Core/Color.js';
import defined from '../Core/defined.js';
import defineProperties from '../Core/defineProperties.js';
```

File: Source/Scene/Cesium3DTileset.js [hunk 1]
```diff
@@ -879,6 +879,19 @@ import TileOrientedBoundingBox from './TileOrientedBoundingBox.js';
     }
 
     defineProperties(Cesium3DTileset.prototype, {
+
```
Comment (lilleyse) [inline Source/Scene/Cesium3DTileset.js]: Style: remove empty line

File: Source/Scene/Cesium3DTileset.js [hunk 2]
```diff
@@ -879,6 +879,19 @@ import TileOrientedBoundingBox from './TileOrientedBoundingBox.js';
...
+        /**
+         * NOTE: This getter exists so that `Picking.js` can differentiate between
+         *       PrimitiveCollections and Cesium3DTileset objects without inflating
+         *       the size of the module via `instance of Cesium3DTileset`
+         */
+
```
Comment (lilleyse) [inline Source/Scene/Cesium3DTileset.js]: Style: remove empty line

File: Source/Scene/Cesium3DTileset.js [hunk 3]
```diff
@@ -879,6 +879,19 @@ import TileOrientedBoundingBox from './TileOrientedBoundingBox.js';
     }
 
     defineProperties(Cesium3DTileset.prototype, {
+
+        /**
+         * NOTE: This getter exists so that `Picking.js` can differentiate between
+         *       PrimitiveCollections and Cesium3DTileset objects without inflating
```
Comment (lilleyse) [inline Source/Scene/Cesium3DTileset.js]:
```suggestion
         *       PrimitiveCollection and Cesium3DTileset objects without inflating
```

File: Source/Scene/Cesium3DTileset.js [hunk 4]
```diff
@@ -879,6 +879,19 @@ import TileOrientedBoundingBox from './TileOrientedBoundingBox.js';
...
+         *       the size of the module via `instance of Cesium3DTileset`
```
Comment (lilleyse) [inline Source/Scene/Cesium3DTileset.js]:
```suggestion
         *       the size of the module via `instanceof Cesium3DTileset`
```

File: Source/Scene/Cesium3DTileset.js [hunk 5]
```diff
@@ -879,6 +879,19 @@ import TileOrientedBoundingBox from './TileOrientedBoundingBox.js';
...
+         */
+
+        isCesium3DTileset: {
```
Comment (lilleyse) [inline Source/Scene/Cesium3DTileset.js]: Style:

```suggestion
        isCesium3DTileset : {
```

File: Source/Scene/Cesium3DTileset.js [hunk 6]
```diff
@@ -879,6 +879,19 @@ import TileOrientedBoundingBox from './TileOrientedBoundingBox.js';
...
+            get: function() {
```
Comment (lilleyse) [inline Source/Scene/Cesium3DTileset.js]: Style:

```suggestion
            get : function() {
```

File: Source/Scene/Cesium3DTileset.js [hunk 7]
```diff
@@ -879,6 +879,19 @@ import TileOrientedBoundingBox from './TileOrientedBoundingBox.js';
...
+                return true;
+            }
+        }
```
Comment (lilleyse) [inline Source/Scene/Cesium3DTileset.js]: This is why CI is failing. You can run `npm run eslint` locally which will also catch this.
```suggestion
        },
```

**Comment by lilleyse** [comment]
Also - add an entry to `CHANGES.md`.

**Comment by Samulus** [comment]
@lilleyse fixed

**Comment by lilleyse** [comment]
Can confirm this fix reduces the build size. For future reference these were the steps @mramato suggested to verify this:

* Switch to `master`
* Edit `CesiumViewer.js` to create a minimal demo:

```js
    window.CESIUM_BASE_URL = '../../Source/';

    import {
        CesiumWidget
    } from '../../Source/Cesium.js';

    function main() {
        var widget = new CesiumWidget('cesiumContainer');
    }

    main();
```

* Run `npm run makeZipFile`
* Run `npm run buildApps`
* Check size of `CesiumViewer.js`
* Do the steps above on this branch

With fix: CesiumViewer.js is 1.3 MB
Without fix: CesiumViewer.js is 1.8 MB

**Comment by lilleyse** [comment]
@Samulus there's one style comment that you missed: https://github.com/AnalyticalGraphicsInc/cesium/pull/8532/files#r365949948

After that's fixed I'll go ahead and merge.

File: Source/Scene/Cesium3DTileset.js (hunk 2 above)
Comment (Samulus) [reply Source/Scene/Cesium3DTileset.js]: @lilleyse removed whitespace

File: Source/Scene/Cesium3DTileset.js [hunk 8]
```diff
@@ -879,6 +879,17 @@ import TileOrientedBoundingBox from './TileOrientedBoundingBox.js';
     }
 
     defineProperties(Cesium3DTileset.prototype, {
+        /**
+         * NOTE: This getter exists so that `Picking.js` can differentiate between
+         *       PrimitiveCollection and Cesium3DTileset objects without inflating
+         *       the size of the module via `instanceof Cesium3DTileset`
+         */
+        isCesium3DTileset : {
```
Comment (lilleyse) [inline Source/Scene/Cesium3DTileset.js]: My mistake for not noticing this in the initial review - the getter should be marked as `@private`.

**Comment by Samulus** [comment]
@lilleyse rebased and annotation added

**Comment by lilleyse** [comment]
Little bit of offline miscommunication on my part, I should have been the one to merge the PR. That said, the changes do look good and I was about to merge :+1:

**Pull request merged**